core/options.c

```diff
@@ -135,27 +135,6 @@ typedef struct _option_t {
     option_modifier_t modifier;
 } option_t;
 
-/* Create an array with traits of all the internal options. */
-#ifndef EXPOSE_INTERNAL_OPTIONS
-/* default values for internal options are kept in a separate struct */
-# define OPTION_COMMAND_INTERNAL(type, name, default_value, command_line_option, \
-                                 statement, description, flag, pcache) \
-{ command_line_option, offsetof(options_t, name), sizeof(type), \
-  OPTION_TYPE_##type, pcache, OPTION_MOD_##flag},
-
-# define OPTION_COMMAND(type, name, default_value, command_line_option, \
-                        statement, description, flag, pcache) /* nothing */
-
-/* Read only source for default internal option values and names. No lock needed
- * since never written.
- */
-static const option_t internal_option_traits[] = {
```

Comment:
OK so for builds where internal options are exposed and first-class options, they are inside option_traits[], and when they're not exposed and are just constants, all we need is the default value in default_internal_options, so we don't need internal_option_traits.  Seems to make sense.

However, per the top-level conversation it sounds like there's one more issue somewhere else?